src/storage/src/keyspace.rs

```diff
@@ -88,63 +87,63 @@ impl<S: StateStore> Keyspace<S> {
     }
 
     /// Gets from the keyspace with the `prefixed_key` of given key.
-    /// The returned value is based on a snapshot corresponding to the given `epoch`
+    /// The returned value is based on a snapshot corresponding to the given `epoch`.
     pub async fn get(&self, key: impl AsRef<[u8]>, epoch: u64) -> StorageResult<Option<Bytes>> {
         self.store.get(&self.prefixed_key(key), epoch).await
     }
 
-    /// Scans `limit` keys from the keyspace and get their values. If `limit` is None, all keys of
-    /// the given prefix will be scanned. Note that the prefix of this keyspace will be stripped.
-    /// The returned values are based on a snapshot corresponding to the given `epoch`
+    /// Scans `limit` keys from the keyspace and get their values.
+    /// If `limit` is None, all keys of the given prefix will be scanned.
+    /// The returned values are based on a snapshot corresponding to the given `epoch`.
     pub async fn scan(
         &self,
         limit: Option<usize>,
         epoch: u64,
     ) -> StorageResult<Vec<(Bytes, Bytes)>> {
-        let range = range_of_prefix(&self.prefix);
+        self.scan_with_range::<_, &[u8]>(.., limit, epoch).await
+    }
+
+    /// Scans `limit` keys from the given `range` in this keyspace and get their values.
+    /// Note that the `range` should not be prepended with the prefix of this keyspace.
+    /// If `limit` is None, all keys of the given prefix will be scanned.
+    /// The returned values are based on a snapshot corresponding to the given `epoch`.
```

Comment:
```suggestion
    /// Scans `limit` keys from the given `range` in this keyspace and get their values.
    /// If `limit` is None, all keys of the given prefix will be scanned.
    /// The returned values are based on a snapshot corresponding to the given `epoch`.
    ///
    /// **Note**: the `range` should not be prepended with the prefix of this keyspace.
```
Make the note more noticable and the remaining the same as `scan` 😁